examples/echo.ts

```diff
@@ -22,10 +22,10 @@ const main = fx(function* () {
 const handlePrint = handle(Print, s => ok(console.log(s)))
 
 const handleRead = <E, A>(f: Fx<E, A>) => bracket(
-  assertSync(() => createInterface({ input: process.stdin, output: process.stdout })),
+  trySync(() => createInterface({ input: process.stdin, output: process.stdout })).pipe(Fail.assert),
   readline => ok(readline.close()),
   readline => f.pipe(
-    handle(Read, prompt => Async.assertPromise(signal => readline.question(prompt, { signal })))
+    handle(Read, prompt => Async.tryPromise(signal => readline.question(prompt, { signal })).pipe(Fail.assert))
```

Comment:
I'm not sure it's all that much better, but couldn't `Fail.assert` be provided here just once `bracket(...).pipe(Fail.assert)`?

Reply:
Yeah, I think you're right.  I'll update it so we can see what that looks like.

Like the talked about on discord, at least so far, this PR doesn't seem like an obvious improvement.  So, I might end up just closing it.

Reply:
Take a look at https://github.com/briancavalier/fx/pull/38/commits/be8481cc0e5b8fa963745b58dba9b75bdc1dbf78 and see what you think of it.